coordinator/coordinator.go

```diff
@@ -276,7 +276,7 @@ func NewCoordinator(cfg Config,
 
 		purger: &purger,
 
-		msgCh: make(chan interface{}),
+		msgCh: make(chan interface{}, 16),
```

Comment:
Maybe leave the size of the buffer in a separated variable and use it in the channel creation. Also, I'd leave a TODO comment on top of it to further analysis to check if the size of the buffer must reside or not in config file.